### PR TITLE
from chatGUIMode, use passive ID checking

### PR DESCRIPTION
**Pull request by maxtaco**
- in passive ID checking, we don't ask for Web and DNS proofs to be resolved *unless* they're already tracked
- this means you can't send a chat to someone and expect them to ping your servera
- no tests for now.

### CODE REVIEW

File: go/libkb/id_table.go [hunk 1]
```diff
@@ -1359,7 +1366,14 @@ func (idt *IdentityTable) proofRemoteCheck(ctx context.Context, hasPreviousTrack
 		return
 	}
 
-	res.err = pc.CheckStatus(idt.G().CloneWithNetContext(ctx), *res.hint)
+	// ProofCheckerModeActive or Passive mainly decides whether we need to reach out to
+	// self-hosted services. We want to avoid so doing when the user is acting passively
+	// (such as when receiving a message).
+	pcm := ProofCheckerModePassive
+	if (hasPreviousTrack && res.trackedProofState != keybase1.ProofState_NONE) || itm == IdentifyTableModeActive {
```
Comment (mlsteele) [inline go/libkb/id_table.go]: to make sure I've got this right: `hasPreviousTrack` is user-specific, not proof-specific. And `_NONE` is the status for never having attempted to check this proof. Is that right?

Comment (maxtaco) [inline go/libkb/id_table.go]: Yes. `hasPreviousTrack` is on if the IDer has tracked the IDee.

`res.trackedProofState != keybase1.ProofState_NONE` if the IDer has previously tracked this component of the IDee.

Comment (mlsteele) [inline go/libkb/id_table.go]: Should probably also make sure it's not `ProofStatus_UNCHECKED` before switching to active.

**Comment by mlsteele** [comment]
Alright, looks good. :+1:

**Comment by maxtaco** [comment]
had to back out that change since it doesn't compile. There's no state that corresponds to "didn't check".  If turned into a `ProofState`, we'd get `ProofState_TEMP_FAILURE` via [this code](https://github.com/keybase/client/blob/b7f56c0afd34ce10cb60c127eabe099415856427/go/libkb/errors.go#L43-L44).

**Comment by maxtaco** [comment]
now, i could make a `ProofState` type and plumb it all the way through correctly....

**Comment by maxtaco** [comment]
... I'll work on it tonight....

**Pull request merged**